### PR TITLE
P4-2213 changes to support password protection of spreadsheet a long with some general tidy up.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Changes to enable password protection of outputted calculated price spreadsheet.

There are some other minor refactoring's also included as part of this change.

Additionally included are some helm changes by @minglis for the secrets config.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/GeoameyPricesProvider.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import java.io.InputStream
 /**
  * Responsible for providing the Geoamey prices Excel spreadsheet via an [InputStream].
  */
-fun interface GeoamyPricesProvider {
+fun interface GeoameyPricesProvider {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/GeoameyPricesProvider.kt:8]: Was mispelt.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/ImporterConfiguration.kt [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
         return JCPTemplateProvider { resourceLoader.getResource(templateFileLocation).inputStream }
     }
 
     @Bean
-    fun timeSource(): TimeSource {
-        return TimeSource { LocalDateTime.now(clock()) }
+    fun timeSource() = TimeSource { LocalDateTime.now(Clock.systemDefaultZone()) }
+
+    @Bean
+    fun spreadsheetProtection(@Value("\${SPREADSHEET_PASSWORD:}") value: String?): SpreadsheetProtection {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/config/ImporterConfiguration.kt:31]: @mafonso FYI

**Review by mafonso — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportController.kt [hunk 3]
```diff
@@ -5,40 +5,45 @@ import org.springframework.format.annotation.DateTimeFormat
...
+            @RequestParam(name = "moves_to", required = true) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) movesTo: LocalDate,
+            @RequestParam(name = "reports_to", required = true) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) reportsTo: LocalDate,
             response: HttpServletResponse?): ResponseEntity<InputStreamResource?>? {
 
         return importService.spreadsheet(supplier, movesFrom, movesTo, reportsTo)?.let { file ->
             val uploadDateTime = timeSource.dateTime().format(DateTimeFormatter.ofPattern("YYYY-MM-dd_HH_mm"))
             val filename = "Journey_Variable_Payment_Output_${supplier}_${uploadDateTime}.xlsx"
             val mediaType: MediaType = MediaType.parseMediaType("application/vnd.ms-excel")
-            val resource = InputStreamResource(FileInputStream(file))
+            val protectedFile = spreadsheetProtection.protectAndGet(file)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/ImportController.kt:40]: @jcavell so was in two minds where to put (call) this.  Decided here in the end over putting in the service.  Can move if want.

**Review by jcavell — APPROVED** [review]

**Pull request merged**